tests/rules/test_StackNameMatchesRegexRule.py

```diff
@@ -0,0 +1,60 @@
+import pytest
+from pycfmodel.model.cf_model import CFModel
+
+from cfripper.config.config import Config
+from cfripper.rules import StackNameMatchesRegexRule
+
+
+@pytest.mark.parametrize(
+    "stack_name, expected_result",
+    [
+        ("justlowercase", True),
+        ("lowercase-with-hyphens", True),
+        ("lowercaseANDUPPERCASE", True),
+        ("lowercase-AND-UPPERCASE-with-hyphens", True),
+        ("also-123-including-456-numbers", True),
+        ("including_underscore", False),
+        ("including space", False),
+        ("including-other-symbols!@£$%^&*()", False),
+    ],
+)
+def test_stack_name_matches_regex(stack_name, expected_result):
+    rule = StackNameMatchesRegexRule(Config(stack_name=stack_name, rules=["StackNameMatchesRegexRule"]))
+    assert rule._stack_name_matches_regex(stack_name) == expected_result
+
+
+def test_works_with_extras():
+    rule = StackNameMatchesRegexRule(Config(stack_name="some-valid-stack-name", rules=["StackNameMatchesRegexRule"]))
+    extras = {"stack": {"tags": [{"key": "project", "value": "some_project"}]}}
+    result = rule.invoke(cfmodel=CFModel(), extras=extras)
+    assert result.valid
+
+
+def test_stack_name_from_extras():
+    rule = StackNameMatchesRegexRule(Config(stack_name="some-valid-stack-name", rules=["StackNameMatchesRegexRule"]))
+    extras = {"stack": {"tags": [{"key": "project", "value": "some_project"}]}, "stack_name": "some_invalid_name"}
+    result = rule.invoke(cfmodel=CFModel(), extras=extras)
+    assert result.valid
+
+
+def test_failure_is_added_for_invalid_stack_name():
```

Comment:
I would be more explicit with this rule, if the reason is going to be hardcoded as (only alphanumerical characters and hyphens allow.) that should be specified in the test name in my opinion.

Reply:
We will provide the reason as an attribute as you have suggested, so we will keep the name.